src/she/sdl2/sdl2_display.cpp

```diff
@@ -0,0 +1,363 @@
+    // case kLinkCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_LINK); break;
+    case kSizeNSCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENS); break;
+    case kSizeWECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEWE); break;
+    // case kSizeNCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEN); break;
+    // case kSizeNECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENE); break;
+    // case kSizeECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEE); break;
+    // case kSizeSECursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZESE); break;
+    // case kSizeSCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZES); break;
+    // case kSizeSWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZESW); break;
+    // case kSizeWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZEW); break;
+    // case kSizeNWCursor: m_cursor = SDL_CreateSystemCursor(SDL_SYSTEM_CURSOR_SIZENW); break;
+    default: break;
+    }
+
+    SDL_SetCursor(m_cursor);
+    m_nativeCursor = cursor;
+    return m_cursor ? true : false;
+  }
+
+  void SDL2Display::setMousePosition(const gfx::Point& position)
+  {
+    SDL_WarpMouseInWindow(
+      m_window,
+      m_scale * position.x,
+      m_scale * position.y);
+  }
+
+  void SDL2Display::captureMouse()
+  {
+    SDL_CaptureMouse(SDL_TRUE);
+  }
+
+  void SDL2Display::releaseMouse()
+  {
+    SDL_CaptureMouse(SDL_FALSE);
+  }
+
+  std::string SDL2Display::getLayout()
+  {
+// #ifdef _WIN32
+//   WINDOWPLACEMENT wp;
+//   wp.length = sizeof(WINDOWPLACEMENT);
+//   if (GetWindowPlacement((HWND)nativeHandle(), &wp)) {
+//     std::ostringstream s;
+//     s << 1 << ' '
+//       << wp.flags << ' '
+//       << wp.showCmd << ' '
+//       << wp.ptMinPosition.x << ' '
+//       << wp.ptMinPosition.y << ' '
+//       << wp.ptMaxPosition.x << ' '
+//       << wp.ptMaxPosition.y << ' '
+//       << wp.rcNormalPosition.left << ' '
+//       << wp.rcNormalPosition.top << ' '
+//       << wp.rcNormalPosition.right << ' '
+//       << wp.rcNormalPosition.bottom;
+//     return s.str();
+//   }
+// #endif
+    return "";
+  }
+
+  void SDL2Display::setLayout(const std::string& layout)
+  {
+// #ifdef _WIN32
+
+//   WINDOWPLACEMENT wp;
+//   wp.length = sizeof(WINDOWPLACEMENT);
+
+//   std::istringstream s(layout);
+//   int ver;
+//   s >> ver;
+//   if (ver == 1) {
+//     s >> wp.flags
+//       >> wp.showCmd
+//       >> wp.ptMinPosition.x
+//       >> wp.ptMinPosition.y
+//       >> wp.ptMaxPosition.x
+//       >> wp.ptMaxPosition.y
+//       >> wp.rcNormalPosition.left
+//       >> wp.rcNormalPosition.top
+//       >> wp.rcNormalPosition.right
+//       >> wp.rcNormalPosition.bottom;
+//   }
+//   else
+//     return;
+
+//   if (SetWindowPlacement((HWND)nativeHandle(), &wp)) {
+//     // TODO use the return value
+//   }
+// #else
+//   // Do nothing
+// #endif
```

Comment:
This should also be implemented in a cross-platform way.

Reply:
Alternatively, this comment should be removed and this function should be a no-op.